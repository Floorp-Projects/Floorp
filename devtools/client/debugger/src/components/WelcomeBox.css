/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

.welcomebox {
  position: absolute;
  top: var(--editor-header-height);
  left: 0;
  bottom: var(--editor-footer-height);
  width: calc(100% - 1px);
  padding: 50px 0 0 0;
  text-align: center;
  background-color: var(--theme-toolbar-background);
  overflow: hidden;
  font-weight: 300;
  z-index: 10;
  user-select: none;
}

.theme-dark .welcomebox {
  background-color: var(--theme-body-background);
}

.alignlabel {
  display: flex;
  white-space: nowrap;
  font-size: 1.25em;
}

.shortcutKeys {
  font-family: Courier;
}

.shortcutKey,
.shortcutLabel {
  flex: 1;
  overflow: hidden;
  text-overflow: ellipsis;
  cursor: pointer;
}

.welcomebox__searchSources:hover,
.welcomebox__searchProject:hover,
.welcomebox__allShortcuts:hover {
  color: var(--theme-body-color);
}

.shortcutKey {
  direction: ltr;
  text-align: right;
  padding-right: 10px;
  font-family: var(--monospace-font-family);
  font-size: 14px;
  line-height: 18px;
  color: var(--theme-body-color);
}

.shortcutKey:dir(rtl) {
  text-align: left;
}

:root[platform="mac"] .welcomebox .shortcutKey {
  font-family: system-ui, -apple-system, sans-serif;
  font-weight: 500;
}

.shortcutLabel {
  text-align: start;
  padding-left: 10px;
  font-size: 14px;
  line-height: 18px;
}

.shortcutFunction {
  margin: 0 auto;
  color: var(--theme-comment);
  display: table;
}

.shortcutFunction p {
  display: table-row;
}

.shortcutFunction .shortcutKey,
.shortcutFunction .shortcutLabel {
  padding: 10px 5px;
  display: table-cell;
}
