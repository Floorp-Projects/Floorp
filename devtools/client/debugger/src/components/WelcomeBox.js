/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

import React, { Component } from "react";

import { connect } from "../utils/connect";

import actions from "../actions";
import { getPaneCollapse } from "../selectors";
import { formatKeyShortcut } from "../utils/text";

import "./WelcomeBox.css";

export class WelcomeBox extends Component {
  render() {
    const searchSourcesShortcut = formatKeyShortcut(
      L10N.getStr("sources.search.key2")
    );

    const searchProjectShortcut = formatKeyShortcut(
      L10N.getStr("projectTextSearch.key")
    );

    const allShortcutsShortcut = formatKeyShortcut(
      L10N.getStr("allShortcut.key")
    );

    const allShortcutsLabel = L10N.getStr("welcome.allShortcuts");
    const searchSourcesLabel = L10N.getStr("welcome.search2").substring(2);
    const searchProjectLabel = L10N.getStr("welcome.findInFiles2").substring(2);
    const { setActiveSearch, openQuickOpen, toggleShortcutsModal } = this.props;

    return (
      <div className="welcomebox">
        <div className="alignlabel">
          <div className="shortcutFunction">
            <p
              className="welcomebox__searchSources"
              role="button"
              tabIndex="0"
              onClick={() => openQuickOpen()}
            >
              <span className="shortcutKey">{searchSourcesShortcut}</span>
              <span className="shortcutLabel">{searchSourcesLabel}</span>
            </p>
            <p
              className="welcomebox__searchProject"
              role="button"
              tabIndex="0"
              onClick={setActiveSearch.bind(null, "project")}
            >
              <span className="shortcutKey">{searchProjectShortcut}</span>
              <span className="shortcutLabel">{searchProjectLabel}</span>
            </p>
            <p
              className="welcomebox__allShortcuts"
              role="button"
              tabIndex="0"
              onClick={() => toggleShortcutsModal()}
            >
              <span className="shortcutKey">{allShortcutsShortcut}</span>
              <span className="shortcutLabel">{allShortcutsLabel}</span>
            </p>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  endPanelCollapsed: getPaneCollapse(state, "end"),
});

export default connect(mapStateToProps, {
  togglePaneCollapse: actions.togglePaneCollapse,
  setActiveSearch: actions.setActiveSearch,
  openQuickOpen: actions.openQuickOpen,
})(WelcomeBox);
