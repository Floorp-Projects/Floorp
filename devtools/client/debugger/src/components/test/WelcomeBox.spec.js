/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

import React from "react";
import { shallow } from "enzyme";

import { WelcomeBox } from "../WelcomeBox";

function render(overrides = {}) {
  const props = {
    horizontal: false,
    togglePaneCollapse: jest.fn(),
    endPanelCollapsed: false,
    setActiveSearch: jest.fn(),
    openQuickOpen: jest.fn(),
    toggleShortcutsModal: jest.fn(),
    ...overrides,
  };
  const component = shallow(<WelcomeBox {...props} />);

  return { component, props };
}

describe("WelomeBox", () => {
  it("renders with default values", () => {
    const { component } = render();
    expect(component).toMatchSnapshot();
  });

  it("doesn't render toggle button in horizontal mode", () => {
    const { component } = render({
      horizontal: true,
    });
    expect(component.find("PaneToggleButton")).toHaveLength(0);
  });

  it("calls correct function on searchSources click", () => {
    const { component, props } = render();

    component.find(".welcomebox__searchSources").simulate("click");
    expect(props.openQuickOpen).toHaveBeenCalled();
  });

  it("calls correct function on searchProject click", () => {
    const { component, props } = render();

    component.find(".welcomebox__searchProject").simulate("click");
    expect(props.setActiveSearch).toHaveBeenCalled();
  });

  it("calls correct function on allShotcuts click", () => {
    const { component, props } = render();

    component.find(".welcomebox__allShortcuts").simulate("click");
    expect(props.toggleShortcutsModal).toHaveBeenCalled();
  });
});
