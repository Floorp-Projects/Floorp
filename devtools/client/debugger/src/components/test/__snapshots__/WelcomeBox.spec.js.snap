// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelomeBox renders with default values 1`] = `
<div
  className="welcomebox"
>
  <div
    className="alignlabel"
  >
    <div
      className="shortcutFunction"
    >
      <p
        className="welcomebox__searchSources"
        onClick={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="shortcutKey"
        >
          Ctrl+P
        </span>
        <span
          className="shortcutLabel"
        >
           Go to file
        </span>
      </p>
      <p
        className="welcomebox__searchProject"
        onClick={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="shortcutKey"
        >
          Ctrl+Shift+F
        </span>
        <span
          className="shortcutLabel"
        >
           Find in files
        </span>
      </p>
      <p
        className="welcomebox__allShortcuts"
        onClick={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="shortcutKey"
        >
          Ctrl+/
        </span>
        <span
          className="shortcutLabel"
        >
          Show all shortcuts
        </span>
      </p>
    </div>
  </div>
</div>
`;
